From b64f96146290d4d7fbf1eb354a06d19baddd5403 Mon Sep 17 00:00:00 2001
From: Igor Romanov <igorro.mlist@gmail.com>
Date: Wed, 7 Oct 2009 02:05:09 +0400
Subject: [PATCH] wxGTK implementation of wxTextCtrl::GetStyle

Getting background colour, text colour and font
is provided for now.
---
 src/gtk/textctrl.cpp |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/src/gtk/textctrl.cpp b/src/gtk/textctrl.cpp
index a29fc54..df56870 100644
--- a/src/gtk/textctrl.cpp
+++ b/src/gtk/textctrl.cpp
@@ -1717,7 +1717,14 @@ bool wxTextCtrl::GetStyle(long position, wxTextAttr& style)
         style.SetBackgroundColour(pattr->appearance.bg_color);
         style.SetTextColour(pattr->appearance.fg_color);
 
-        // TODO: set font, alignment, tabs and indents
+        char *pangoFontString = pango_font_description_to_string(pattr->font);
+        wxON_BLOCK_EXIT1( g_free, pangoFontString );
+
+        wxFont font;
+        if ( font.SetNativeFontInfo( wxString(pangoFontString) ) )
+            style.SetFont(font);
+
+        // TODO: set alignment, tabs and indents
     }
 
     return true;
-- 
1.6.2.5

